### PR TITLE
pre-release v0.3.0

### PR DESCRIPTION
**Pull request by tstescoTT**
## Summary of Changes

**vLLM server**
- added the signal file to enable k8s probes to detect the warmup completion https://github.com/tenstorrent/tt-inference-server/pull/1014

**tt-media-server**
- SD 3.5 accuracy checks and performance improvements

**System SW (applies to Wormhole Galaxy):**
- tt-smi: 3.0.31
- Firmware: 18.12.0
- KMD: 2.4.1

# Model Spec Release Updates

This document shows model specification updates.

| Impl | Model Arch | Weights | Devices | TT-Metal Commit Change | Status Change | CI Job Link |
|------|------------|---------|---------|------------------------|---------------|-------------|
| `tt_transformers` | `Qwen3-8B` | `Qwen/Qwen3-8B` | T3K, GALAXY_T3K, GALAXY | `2496be4` → `13f44c5` | FUNCTIONAL (no change) | [Run 21](https://github.com/tenstorrent/tt-shield/actions/runs/18992172929/job/54264525379) |
| `tt_transformers` | `Qwen3-32B` | `Qwen/Qwen3-32B` | T3K, GALAXY_T3K, GALAXY | `2496be4` → `13f44c5` | FUNCTIONAL (no change) | [Run 21](N/A) |
| `tt_transformers` | `QwQ-32B` | `Qwen/QwQ-32B` | T3K, GALAXY, GALAXY_T3K | `2496be4` → `13f44c5` | FUNCTIONAL (no change) | [Run 21](https://github.com/tenstorrent/tt-shield/actions/runs/18992172929/job/54264525439) |
| `tt_transformers` | `Qwen2.5-72B` | `Qwen/Qwen2.5-72B`<br>`Qwen/Qwen2.5-72B-Instruct` | T3K, GALAXY, GALAXY_T3K | `2496be4` → `13f44c5` | COMPLETE → FUNCTIONAL | [Run 21](N/A) |
| `llama3_70b_galaxy` | `Llama-3.3-70B-Instruct` | `meta-llama/Llama-3.3-70B-Instruct`<br>`meta-llama/Llama-3.1-70B`<br>`meta-llama/Llama-3.1-70B-Instruct`<br>`deepseek-ai/DeepSeek-R1-Distill-Llama-70B` | GALAXY | `2496be4` → `13f44c5` | COMPLETE (no change) | [Run 21](N/A) |
| `tt_transformers` | `Llama-3.1-8B` | `meta-llama/Llama-3.1-8B`<br>`meta-llama/Llama-3.1-8B-Instruct` | GALAXY, GALAXY_T3K | `2496be4` → `13f44c5` | FUNCTIONAL (no change) | N/A |
| `tt_transformers` | `stable-diffusion-xl-base-1.0` | `stabilityai/stable-diffusion-xl-base-1.0` | N150, N300, T3K, GALAXY | `2496be4` → `13f44c5` | COMPLETE (no change) | [21](https://github.com/tenstorrent/tt-shield/actions/runs/18992172929/job/54264525380) |
| `tt_transformers` | `stable-diffusion-3.5-large` | `stabilityai/stable-diffusion-3.5-large` | N150, N300, T3K, GALAXY | `2496be4` → `13f44c5` | FUNCTIONAL → COMPLETE | [21](https://github.com/tenstorrent/tt-shield/actions/runs/18992172929/job/54264525421) |

# Model Waivers
1. Qwen3-8B Galaxy: accuracy variance, due to subsampling of eval dataset the eval score drops below 95% to reference. The correct score has been confirmed outside Models CI on full dataset.

# Release Artifacts Summary

## Images Promoted from Models CI

- https://ghcr.io/tenstorrent/tt-inference-server/vllm-tt-metal-src-dev-ubuntu-22.04-amd64:0.3.0-13f44c5-0edd242
  - from: https://ghcr.io/tenstorrent/tt-shield/vllm-tt-metal-src-dev-ubuntu-22.04-amd64:0.2.0-13f44c56cc4927da87567a8358d45151f76b6423-0edd242-54246849490

**Total:** 1

## Existing Images with Models CI reference

Images that already exist on remote and have a valid Models CI image available.

No existing images with Models CI reference.

## Existing Images without Models CI reference

Images that already exist on remote but have no valid Models CI reference (manually built/pushed).

- https://ghcr.io/tenstorrent/tt-media-inference-server:0.3.0-13f44c56cc4927da87567a8358d45151f76b6423

**Total:** 1

## Docker Images Requiring New Builds

**Note:** Model Specs added outside of Models CI will need to have Docker images built manually and will show up here if not already existing. This will happen by design when a release happens and the VERSION file is incremented.

- https://ghcr.io/tenstorrent/tt-inference-server/vllm-tt-metal-src-dev-ubuntu-22.04-amd64:0.3.0-17a5973-aa4ae1e
- https://ghcr.io/tenstorrent/tt-inference-server/vllm-tt-metal-src-dev-ubuntu-22.04-amd64:0.3.0-20edc39-03cb300
- https://ghcr.io/tenstorrent/tt-inference-server/vllm-tt-metal-src-dev-ubuntu-22.04-amd64:0.3.0-55fd115-aa4ae1e
- https://ghcr.io/tenstorrent/tt-inference-server/vllm-tt-metal-src-dev-ubuntu-22.04-amd64:0.3.0-5b5db8a-e771fff
- https://ghcr.io/tenstorrent/tt-inference-server/vllm-tt-metal-src-dev-ubuntu-22.04-amd64:0.3.0-9b67e09-a91b644
- https://ghcr.io/tenstorrent/tt-inference-server/vllm-tt-metal-src-dev-ubuntu-22.04-amd64:0.3.0-ae65ee5-35f023f
- https://ghcr.io/tenstorrent/tt-inference-server/vllm-tt-metal-src-dev-ubuntu-22.04-amd64:0.3.0-dc85f59-87fe4a4
- https://ghcr.io/tenstorrent/tt-inference-server/vllm-tt-metal-src-dev-ubuntu-22.04-amd64:0.3.0-v0.61.1-rc1-5cbc982
- https://ghcr.io/tenstorrent/tt-inference-server/vllm-tt-metal-src-dev-ubuntu-22.04-amd64:0.3.0-v0.62.0-rc33-e7c329b

**Total:** 9

## vLLM feature support table

| Section / Parameter | Llama-3.3-70B/ 8B-Instruct (TT) | Llama-3.3-70B/ 8B-Instruct (TT Compat) | Qwen3-32B (TT) | Qwen3-32B (TT Compat) | QWQ-32B (TT) | QWQ-32B (TT Compat) | Qwen2.5-72B-Instruct (TT) | Qwen2.5-72B-Instruct (TT Compat) |
|:---|:---|:---|:---|:---|:---|:---|:---|:---|
| **/v1/chat/completions parameters** | | | | | | | | |
| n | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ |
| best\_of | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ |
| presence\_penalty | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| frequency\_penalty | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| repetition\_penalty | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| temperature | ✅ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| temperature | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| messages | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
| top\_p | ✅ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| top\_p | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| top\_k | ✅ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| min\_p | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| reasoning\_effort | | | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ |
| seed | ✅ (doesn't do anything) | ✅ | ✅ (doesn't do anything) | ✅ | ✅ (doesn't do anything) | ✅ | ✅ (doesn't do anything) | ✅ |
| stop | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
| stop\_token\_ids | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
| bad\_words | ❌ | ❌ (bug - crashes server) | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| include\_stop\_str\_in\_output | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
| ignore\_eos | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
| max\_tokens | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
| min\_tokens | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
| logprobs | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| prompt\_logprobs | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ |
| detokenize | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
| skip\_special\_tokens | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
| spaces\_between\_special\_tokens | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
| logits\_processors | ❌ | ✅* (this param is only used with offline vLLM engine, not through server) | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ |
| truncate\_prompt\_tokens | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| output\_kind | ❌ | ✅* (this param is only used with offline vLLM engine, not through server) | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ |
| structured\_outputs | ❌ | ✅ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ |
| guided\_decoding | ❌ | ✅ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ |
| logit\_bias | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| allowed\_token\_ids | ❌ | ❌ (bug - crashes server) | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| extra\_args | ✅ (only some args, no guided-anything) | ✅ | ✅ (only some args, no guided-anything) | ✅ (only some args, no guided-anything) | ✅ (only some args, no guided-anything) | ✅ (only some args, no guided-anything) | ✅ (only some args, no guided-anything) | ✅ (only some args, no guided-anything) |
| **Supporting Functions** | | | | | | | | |
| Reasoning | ❌ | ❌ | ✅ | ✅ | ✅ | ✅ | ✅ | ❌ |
| Function/Tool Calling | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
| Structured Outputs | ❌ | ✅ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ |
| Browser Search | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ |
| Code Execution | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ |
| Image input | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ |
| Files input | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ |
| Code Generation | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ |
| Code Reasoning | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ |
| Code Fixing | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ |
| **Model Specific parameters** | | | | | | | | |
| stream | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
| stream | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| stream\_options | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
| response\_format | ❌ | ✅ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ |
| tools | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
| tools | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| tools\_choice | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
| **/v1/responses parameters** | | | | | | | | |
| background | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| include | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| input | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| instructions | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| max\_output\_tokens | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| max\_tool\_calls | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| metadata | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| model | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| parallel\_tool\_calls | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| previous\_response\_id | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| prompt | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| reasoning | ❌ | ✅ (doesn't do anything) | ❌ | ✅ | ❌ | ✅ | ❌ | ❌ |
| service\_tier | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| store | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| stream | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
| stream | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| temperature | ✅ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| temperature | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| text | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ |
| tool\_choice | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| tools | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ | ✅ |
| tools | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| top\_logprobs | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ |
| top\_p | ✅ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| top\_p | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| truncation | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| user | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| request\_id | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| mm\_processor\_kwargs | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ | ❌ | ✅ |
| priority | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ |
| cache\_salt | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ | ❌ |

#### Notes:
1. TT Compat corresponds to configuring `--vllm-override-args "{"always_compat_sampling": true, "sample_on_device_mode": null}"`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,8 +38,8 @@ For automated and pre-configured vLLM inference server using Docker please see t
...
+| [stable-diffusion-xl-base-1.0](https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0) | [Galaxy](https://tenstorrent.com/hardware/galaxy), [n300](https://tenstorrent.com/hardware/wormhole), [WH-QuietBox](https://tenstorrent.com/hardware/tt-quietbox)/[WH-LoudBox](https://tenstorrent.com/hardware/tt-loudbox) (T3K), [n150](https://tenstorrent.com/hardware/wormhole) | 🟢 Complete | [13f44c5](https://github.com/tenstorrent/tt-metal/tree/13f44c5/models/tt_transformers) | N/A | [0.3.0-13f44c56cc4927da87567a8358d45151f76b6423](https://ghcr.io/tenstorrent/tt-media-inference-server:0.3.0-13f44c56cc4927da87567a8358d45151f76b6423) |
+| [stable-diffusion-3.5-large](https://huggingface.co/stabilityai/stable-diffusion-3.5-large) | [WH-QuietBox](https://tenstorrent.com/hardware/tt-quietbox)/[WH-LoudBox](https://tenstorrent.com/hardware/tt-loudbox) (T3K), [Galaxy](https://tenstorrent.com/hardware/galaxy) | 🛠️ Experimental | [13f44c5](https://github.com/tenstorrent/tt-metal/tree/13f44c5/models/tt_transformers) | N/A | [0.3.0-13f44c56cc4927da87567a8358d45151f76b6423](https://ghcr.io/tenstorrent/tt-media-inference-server:0.3.0-13f44c56cc4927da87567a8358d45151f76b6423) |
```
Comment (idjuricTT) [inline README.md]: SD-3.5 should be complete as well
